src/shared/hooks/use-property-update.ts

```diff
@@ -0,0 +1,18 @@
+import usePrevious from "react-hooks-use-previous";
+import { useState } from "react";
+
+const loading = "loading";
+
+// A simple hook that accepts object and a property name and returns `true` when the property has been updated.
+// There's an assumption that the object can be initially unavailable (undefined) and this should not affect
+// the result. This hook will detect change of the property, not property and object combination.
+// It's useful for observing interactive state properties.
+export const usePropertyUpdate = (object: any, propertyName: string) => {
```

Comment:
I found the name `usePropertyUpdate` less than transparent on first reading. How would you feel about `usePropertyDidChange` or some variant thereof?

Reply:
I renamed it, thanks. All the naming suggestions are welcome. ;)